src/main.rs

```diff
@@ -269,20 +191,51 @@ fn main() {
     } else {
         command_strings.map(|c| Command::new(c)).collect()
     };
+    commands
+}
 
-    let res = run(&commands, &options);
-
-    match res {
-        Ok(timing_results) => {
-            write_benchmark_comparison(&timing_results);
-            let ans = export_manager.write_results(timing_results);
-            if let Err(e) = ans {
-                error(&format!(
-                    "The following error occured while exporting: {}",
-                    e.description()
-                ));
+macro_rules! test_build_commands_ranges {
+    ($name:ident, $v:expr, $e:expr) => {
+        #[test]
+        fn $name() {
+            let a = $v;
+            let actual = build_commands(&a);
+            assert_eq!($e.len(), actual.len());
+            for (_, (exp, act)) in $e.iter().zip(actual.iter()).enumerate() {
+                assert_eq!(exp.get_shell_command(), act.get_shell_command());
             }
         }
-        Err(e) => error(e.description()),
     }
 }
+
+test_build_commands_ranges!(
+    both_positive,
+    build_arg_matches_for_commands(vec![
+        "hyperfine",
+        "ls -a {val}",
+        "--parameter-scan",
+        "val",
+        "0",
+        "5",
+    ]),
+    build_commands_over_range("ls -a {val}", "val", 0..5)
+);
+
+#[cfg(test)]
+fn build_arg_matches_for_commands(values: Vec<&'static str>) -> ArgMatches<'static> {
+    use std::ffi::OsString;
+    get_arg_matches(values.into_iter().map(|v| OsString::from(v)))
+}
+
+#[cfg(test)]
+fn build_commands_over_range(
```

Comment:
It somehow feels wrong that we have to basically duplicate the functionality in order to test it. In my opinion, unit tests should be self-contained with hard-coded expected results.

Reply:
This was just done to avoid writing the same logic over and over again. We could just build the expected commands for each test (there should be a few possible ones for this particular function), which is what I originally did, but that becomes rather verbose.